File: packages/rune-secure-store/ios/RuneSecureStore.swift
//
//  RuneSecureStore.swift
//  RuneSecureStore
//
//  Public interface for RuneSecureStore module
//

import Foundation
import RuneKit

@objc(RuneSecureStore)
public class RuneSecureStore: NSObject {
  private static var moduleInstance: RuneSecureStoreModule?

  @objc public static func initialize(with runtime: RuneRuntime) {
    guard moduleInstance == nil else {
      print("[RuneSecureStore] Module already initialized")
      return
    }

    let module = RuneSecureStoreModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[RuneSecureStore] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/rune-secure-store/android/RuneSecureStore/src/main/java/dev/rune/securestore/RuneSecureStore.kt
package dev.rune.securestore

import android.app.Activity
import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneSecureStore module.
 */
object RuneSecureStore {
    private var moduleInstance: RuneSecureStoreModule? = null

    /**
     * Initialize the SecureStore module with an Activity and RuneRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: RuneRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("RuneSecureStore", "Module already initialized")
            return
        }

        moduleInstance = RuneSecureStoreModule(activity)
        runtime.installModules(listOf(moduleInstance!!))
        android.util.Log.d("RuneSecureStore", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}
